framework/src/vectorpostprocessors/SamplerBase.C

```diff
@@ -87,6 +89,33 @@ SamplerBase::initialize()
       _values.begin(), _values.end(), [](VectorPostprocessorValue * vec) { vec->clear(); });
 }
 
+void
+SamplerBase::checkSampleStandardFieldVariableType(const MooseVariableFieldBase * const var_ptr,
+                                                  const std::string & var_param_name) const
+{
+  // A pointer to a MooseVariableFieldBase will likely never be SCALAR
+  mooseAssert(var_ptr->feType().family != SCALAR,
+              "Scalar variable '" + var_ptr->name() + "' cannot be sampled.");
+  mooseAssert(dynamic_cast<const MooseObject *>(_vpp), "Should have succeeded");
+  if (var_ptr->isVector())
+    dynamic_cast<const MooseObject *>(_vpp)->paramError(
```

Comment:
I wish we could use _vpp directly here, but VPP is not bringing the MooseObject type, the UO-class is. 
This would do? We could static cast too but we dont need the performance for an error message

Reply:
turns out, cant do it
```
/data/civet2/build/moose/framework/src/vectorpostprocessors/SamplerBase.C:110:5: error: static_cast from 'VectorPostprocessor *' to 'const MooseObject *', which are not related by inheritance, is not allowed
    static_cast<const MooseObject *>(_vpp)->paramError(
```

Reply:
oh yea forgot only `dynamic_cast` will allow you to cast sideways